### PR TITLE
[ContactStructuralMechanicsApplication] Separating the contact search into simple/advanced

### PR DESCRIPTION
**Pull request by loumalouomega**
This simplifies internally some operations

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/custom_utilities/base_contact_search.h [hunk 1]
```diff
@@ -133,12 +135,12 @@ class TreeContactSearch
...
      *          -# InterfacePart = InterfaceMapper.GenerateInterfacePart(Complete_Model_Part)
      */
-    TreeContactSearch( 
+    BaseContactSearch(
         ModelPart& rMainModelPart, 
         Parameters ThisParameters =  Parameters(R"({})") 
         ); 
     
-    virtual ~TreeContactSearch()= default;;
+    virtual ~BaseContactSearch()= default;;
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_utilities/base_contact_search.h:143]:
```suggestion
    virtual ~BaseContactSearch()= default;
```

File: applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_results.json [hunk 2]
```diff
@@ -1 +1 @@
...
... -0.0008627548337711778, -0.000988761650277525, -0.0010519284281549023, -0.001084505311085592, -0.0010285771169496554, -0.0009170636146273136, -0.0006348813534886479, -0.0002652576620199583, 0.00023339758060775232, 0.0008210326488323616, 0.0014653655869448065], "DISPLACEMENT_Y": [-0.002231468399327593, -0.0020397848849805986, -0.00038064078232221435, 0.0008196165077470075, 0.0012279863050812696, 0.0013996189167086413, 0.001460221964200122, 0.0012848300886438243, 0.001109038791897729, 0.0011134988098398608, 0.0012933588318270127, 0.00147092288497016, 0.0014199124467222054, 0.0012555890656368162, 0.0008621068282691907, -0.00030272839491851227, -0.0019399885570926916]}, "NODE_48": {"DISPLACEMENT_X": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], "DISPLACEMENT_Y": [0.0012269474934462879, 0.0012979233675630033, 0.001848103087485157, 0.002141565701608695, 0.001937831430995609, 0.0016927630471475807, 0.0014490083550660914, 0.0011914925466656248, 0.0009717058200175363, 0.0009744197600855788, 0.0011972506844963227, 0.001455896561210448, 0.0016986145019140262, 0.0019435822859551017, 0.0021568582136395615, 0.0018859595692014626, 0.0013502642510822246]}}
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_results.json:1]: why do the results change if you only refactor?

**Review by philbucher — APPROVED** [review]
three comments:
- there is quite some commented code
- what is the difference btw simple and advanced search? => I think the class descriptions for those are not up-to-date
- the templates seems to make your life harder and harder ...

otherwise not much to say, your app so I guess you know what you are doing :)

File: applications/ContactStructuralMechanicsApplication/custom_utilities/base_contact_search.h (hunk 1 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_utilities/base_contact_search.h:143]: I will upload the changes in a future PR. I found some minor errors on OMP these days

File: applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_results.json (hunk 2 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_results.json:1]: I changed some defaults

**Pull request merged**